src/sdk/namespace_fs.js

```diff
@@ -199,6 +215,55 @@ const dir_cache = new LRUCache({
     max_usage: config.NSFS_DIR_CACHE_MAX_TOTAL_SIZE,
 });
 
+/**
+ * @typedef {{
+ *  time: number,
+ *  stat: fs.Stats,
+ *  usage: number,
+ *  sorted_entries?: fs.Dirent[],
+ * }} ReaddirVersionsCacheItem
+ * @type {LRUCache<object, string, ReaddirCacheItem>}
+ */
+const versions_dir_cache = new LRUCache({
+    name: 'nsfs-versions-dir-cache',
+    make_key: ({ dir_path }) => dir_path,
+    load: async ({ dir_path, fs_context }) => {
+        const time = Date.now();
+        const stat = await nb_native().fs.stat(fs_context, dir_path);
+        const sorted_entries = [];
+        let usage = config.NSFS_DIR_CACHE_MIN_DIR_SIZE;
+        if (stat.size <= config.NSFS_DIR_CACHE_MAX_DIR_SIZE) {
```

Comment:
you should check for the sum of dir size and ver size

```suggestion
        if (stat.size + ver_dir_stat.size <= config.NSFS_DIR_CACHE_MAX_DIR_SIZE) {
```

Reply:
sure